packages/flame_3d/lib/src/extensions/quaternion.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:math';
+
+import 'package:flame_3d/core.dart';
+
+extension QuaternionExtension on Quaternion {
+  List<double> get storage => [x, y, z, w];
+
+  double dot(Quaternion other) {
+    return x * other.x + y * other.y + z * other.z + w * other.w;
+  }
+
+  Quaternion lerp(Quaternion other, double t) {
+    return QuaternionUtils.lerp(this, other, t);
+  }
+
+  Quaternion slerp(Quaternion other, double t, {double epsilon = 10e-6}) {
+    return QuaternionUtils.slerp(this, other, t, epsilon: epsilon);
+  }
+}
+
+extension QuaternionUtils on Quaternion {
```

Comment:
Does this event have to be an extension? It could just be a `final class`?

Reply:
it can def be a class, let me fix